amy/dashboard/views.py

```diff
@@ -238,25 +269,33 @@ def training_progress(request):
         get_involved_form = GetInvolvedForm(
             data=request.POST, instance=base_training_progress
         )
+
         if get_involved_form.is_valid():
-            get_involved_form.save()
+            return self.form_valid(get_involved_form)
 
-            messages.success(
-                request,
-                "Thank you. Your Get Involved submission will be reviewed within 7-10 "
-                "days.",
-            )
-            return redirect(reverse("training-progress"))
+        else:
+            return self.form_invalid(get_involved_form)
 
-    context = {
-        "title": "Your training progress",
-        "get_involved_form": get_involved_form,
-        "progress_training": progress_training,
-        "progress_get_involved": progress_get_involved,
-        "progress_welcome": progress_welcome,
-        "progress_demo": progress_demo,
-    }
-    return render(request, "dashboard/training_progress.html", context)
+
+class GetInvolvedUpdateView(LoginRequiredMixin, RedirectSupportMixin, AMYUpdateView):
```

Comment:
I think `RedirectSupportMixin` is not needed here, as `ModelFormMixin` (part of Django's `UpdateView`) handles redirect after form submission.